solara/website/pages/examples/visualization/annotator.py

```diff
@@ -0,0 +1,60 @@
+"""# Image annotation with Solara
+
+This example displays how to annotate images with different drawing tools in plotly figures. Use the canvas
+below to draw shapes and visualize the canvas callback.
+
+
+Check [plotly docs](https://dash.plotly.com/annotations) for more information about image annotation.
+"""
+import json
+import plotly.graph_objects as go
+
+import solara
+
+title = "Plotly Image Annotator"
+shapes = solara.reactive(None)
+
+class CustomEncoder(json.JSONEncoder):
+    def default(self, o):
+        # Convert object instances to their string representation
+        if isinstance(o, object):
+            return str(o)
+        return super().default(o)
```

Comment:
Why is this needed? Does plotly give back some objects? If so, it would be good to explain this in this piece of code.

Reply:
> Does plotly give back some objects?

Yes, I added more explanation about it.